### ISSUE TITLE
[feature request]: Little endian functions for setting and getting `u32`

### ISSUE THREAD

**Issue by Eisverygoodletter** [alttch/rmodbus#36]
Currently, the function for `set_holdings_from_u32` and `set_inputs_from_u32` and their `get` equivalents both use a big-endian representation with
```rust
...
(self.inputs)[reg as usize] = (value >> 16) as u16;
(self.inputs)[reg as usize + 1] = value as u16;
...
```
Would you be interested in adding functions `set_holdings_from_u32_le` and `set_inputs_from_u32_le` (and their `get` equivalents) for little endian? For example
```rust
fn set_inputs_from_u32_le(&mut self, reg: u16, value: u32) {
    if reg as usize + 1 >= I {
        Err(ErrorKind::OOBContext)
} else {
    (self.inputs)[reg as usize] = value as u16;
    (self.inputs)[reg as usize + 1] = (value >> 16) as u16;
}
```
and maybe extra functions `set_holdings_from_u32_be` and `set_inputs_from_u32_be` which are identical to the current `set_holdings_from_u32` and `set_inputs_from_u32` implementation for clarity.

I'm happy to implement this and create a pull request if you are interested in adding this.

**Comment by divi255** [comment]
Good day,

this can be discussed, but why don't just use u32::swap_bytes before calling the setter method?

**Comment by Eisverygoodletter** [comment]
That is possible and would be equivalent to implementing a new function, but I think a new function specifically for little endian would be clearer because then I won't have to write comments explaining that the `u32::swap_bytes` call is for little endian every time its used.

**Comment by divi255** [comment]
sounds reasonable, but it should be then implemented for both u32 and u64. I'll think about how to better implement this. the problem in modbus endianess is actually that the registers are u16, some vendors swap all bytes, some swap words. it's pretty hard to implement something really versatile for all cases. I'll think about it, if you have some time, feel free to PR as well

**Comment by Eisverygoodletter** [comment]
Perhaps something like
```rust
// Anything that can be stored in (multiple) registers
pub trait RegisterRepresentable<const N: usize> {
    fn to_registers_sequential(&self) -> [u16; N];
    fn from_registers_sequential(&self, value: &[u16; N]) -> Self;
}
pub struct U32BigEndian(pub u32);
impl RegisterRepresentable<2> for U32BigEndian(pub u32) {}

pub struct U32LittleEndian(pub u32);
impl RegisterRepresentable<2> for U32LittleEndian(pub u32) {...}
```
With
```rust
fn set_inputs_with_representable<const N: usize, T: RegisterRepresentable<N>>(
    &mut self, 
    starting_reg: u16, 
    representable: T
) -> Result<(), ErrorKind> {
    let represented_as_u16 = representable.to_registers_sequential();
    self.set_inputs_bulk(starting_reg, represented_as_u16)
}
```
and its `get` equivalent provided? 

Then users of this library can define their `struct ReallyReallyWeirdU128Representation(...)` and implement `RegisterRepresentable` on it if they need to. This way the library can provide a few simpler ones like `U32BigEndian` and `U32LittleEndian` and if users need it, they can implement more custom structs and representations.

This does cover way more stuff than just `u32` and `u64` compared to the original addition. I can make a PR with this if you'd like. What do you think?

**Comment by divi255** [comment]
yep something like this. it's nearly impossible to describe all representation variations in some config so a trait is the key

**Comment by Eisverygoodletter** [comment]
I'll make a fork and set up a PR when ready.

**Comment by divi255** [comment]
available in 0.9.10
